omniqueue/src/backends/redis/mod.rs

```
@@ -326,6 +370,22 @@ impl<R: RedisConnection> RedisBackendBuilder<R> {
             }
         });
 
+        join_set.spawn({
+            async move {
+                if let Err(e) = check_eviction_policy(redis.clone()).await {
+                    match e {
+                        EvictionCheckError::CheckEvictionPolicyFailed => {
+                            tracing::warn!("{e}");
```

This isn't really telling us why the eviction policy failed.

I think I miscommunicated my original suggestion. I don't really see why a new error variant is helpful here, or why "CheckEvictionPolicyFailed" is a useful error.

It seems like "Unsafe Eviction Policy" should just be part of the normal omniqueue error type. `check_eviction_policy`, if it fails to check the policy, should just report the underlying error of why. And then the caller of `check_eviction_policy` can decide how they want to handle that kind of error.

---

The reason is that I don't want to log an error if it's unable to check the eviction policy -- the failure to do so is fairly common. Note the PR description comment about cloud providers.

I don't think it's worth adding a variant to the common error type, which would be breaking, correct?

---

> I don't think it's worth adding a variant to the common error type, which would be breaking, correct?

Yes, it would. And I really hate that this is forcing us make our codebase worse.

> The reason is that I don't want to log an error if it's unable to check the eviction policy -- the failure to do so is fairly common

Yeah, I got that. But my point is that I don't think we should be making this decision for all omniqueue users. It should be up to whoever initializes the redis backend to decide how they want to handle this error.

---

Maybe the issue is that I'm trying to log this as an `error` at all. The eviction config is an infrastructure concern. It doesn't actually impact the performance of omniqueue, though it may cause data loss in the same way that non-persistent redis might. It is something that bears a warning to the user, though I don't think it's something that should be surfaced as an error when starting the process.